### PR TITLE
Add ubuntu-16.04-ppc64le support in pl-build-tools

### PR DESCRIPTION
**Pull request by ScottGarman**
This builds out our toolchain for the Power8 architecture on Ubuntu 16.04. The main thing you may find counterintuitive is that the gcc host triple for this platform is powerpc64le-linux-gnu, but the Debian package architecture is ppc64el. We do things like name our platform config files based on the packaging architecture, not the compile triple. We also have to fiddle a bit with our sysroot generation to workaround issues with leatherman not yet supporting Debian-style multilib path searching.

### CODE REVIEW

**Comment by McdonaldSeanp** [comment]
👍

**Comment by branan** [comment]
This loooks clean to me :+1:

I think the fiddling we do to make the sysroot cleaner is fine here. Easier than trying ot mess with CMake configs to make it work with a more true-to-the-original sysroot

**Comment by Magisus** [comment]
I don't have a lot of experience reviewing this stuff, but what's here seems to make sense. If I wanted to try this out, what would I need to do? Is that even possible?

File: configs/components/gcc.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 component "gcc" do |pkg, settings, platform|
   # Source-Related Metadata
-  if platform.name =~ /fedora-f24|fedora-f25|ubuntu-16\.10/
+  if platform.name =~ /fedora-f24|fedora-f25|ubuntu-16\.04-ppc64el|ubuntu-16\.10/
```
Comment (geoffnichols) [inline configs/components/gcc.rb]: What are thoughts on revving our pl-gcc version to 6.1 for Ubuntu 16.04 for all arches (not just`ppc64el`)?

Comment (ScottGarman) [inline configs/components/gcc.rb]: A pl-gcc rev for a platform requires rebuilding and re-shipping everything else in pl-build-tools - but I'd describe it as a low-risk rebuild that can be done as a 3-point background task (just to account for context switching load). If you want to go forward with this, I'm not opposed to it.

Comment (geoffnichols) [inline configs/components/gcc.rb]: Thanks @ScottGarman. Sounds like something we may want to revisit separately.

**Comment by ScottGarman** [comment]
@Magisus it would just be a matter of checking out this branch and then building each of the pl-build-tools targets for this platform in order - pl-binutils, pl-gcc, pl-boost, pl-cmake and pl-ruby (for amd64), and pl-yaml-cpp. The subtlety is just that since these packages haven't yet been shipped to our internal repositories, you'd have to scp each of the build artifacts to a pre-checked out build host and between each build, uninstall them and reinstall them (I typically rm -rf /var/tmp/tmp.* as well between builds).

**Comment by Magisus** [comment]
I'm going to file this away for future reference, and assume someone else has done that this go-round :P Thanks for explaining that!

**Pull request merged**